src/domain/Source.js

```diff
@@ -0,0 +1,16 @@
+// @flow
```

Comment:
Can you rename it `DirectorySource` ? It can be confusing with other type of source.